test/services/GetFinancialDetailsServiceSpec.scala

```diff
@@ -0,0 +1,138 @@
+      financialDetails = Seq(FinancialDetails(
+        documentId = "0001",
+        taxPeriodFrom = Some(LocalDate.of(2022, 10 ,30)),
+        taxPeriodTo = Some(LocalDate.of(2022,10,30)),
+        items = Seq(FinancialItem(
+          dueDate = Some(LocalDate.of(2022, 10 ,30)),
+          clearingDate = Some(LocalDate.of(2022, 10 ,30)),
+          metadata = FinancialItemMetadata(
+            subItem = Some("001"),
+            amount = Some(123.45),
+            clearingReason = Some("01"),
+            outgoingPaymentMethod = Some("outgoing Payment"),
+            paymentLock = Some("paymentLock"),
+            clearingLock = Some("clearingLock"),
+            interestLock = Some("interestLock"),
+            dunningLock = Some("dunningLock"),
+            returnFlag = Some(true),
+            paymentReference = Some("Ab12453535"),
+            paymentAmount = Some(543.21),
+            paymentMethod = Some("Payment"),
+            paymentLot = Some("81203010024"),
+            paymentLotItem = Some("000001"),
+            clearingSAPDocument = Some("3350000253"),
+            codingInitiationDate = Some(LocalDate.of(2022, 10 ,30)),
+            statisticalDocument = Some("S"),
+            DDCollectionInProgress = Some(true),
+            returnReason = Some("ABCA"),
+            promisetoPay = Some("promisetoPay")
+          )
+        )),
+        originalAmount = Some(123.45),
+        outstandingAmount = Some(543.21),
+        metadata = FinancialDetailsMetadata(
+          taxYear = "2022",
+          chargeType = Some("PAYE"),
+          mainType = Some("2100"),
+          periodKey = Some("13RL"),
+          periodKeyDescription = Some("abcde"),
+          businessPartner = Some("6622334455"),
+          contractAccountCategory = Some("02"),
+          contractAccount = Some("X"),
+          contractObjectType = Some("ABCD"),
+          contractObject = Some("00000003000000002757"),
+          sapDocumentNumber = Some("1040000872"),
+          sapDocumentNumberItem = Some("XM00"),
+          chargeReference = Some("XM002610011594"),
+          mainTransaction = Some("1234"),
+          subTransaction = Some("5678"),
+          clearedAmount = Some(123.45),
+          accruedInterest = Some(543.21)
+        )
+      ))
+    )
+
+    s"call the connector and return a $GetFinancialDetailsSuccessResponse when the request is successful" in new Setup {
+      when(mockGetFinancialDetailsConnector.getFinancialDetails(Matchers.eq("123456789"), Matchers.eq(LocalDate.now()), Matchers.eq(LocalDate.now()))(any()))
+        .thenReturn(Future.successful(Right(GetFinancialDetailsSuccessResponse(mockGetFinancialDetailsResponseAsModel))))
+
+      val result: GetFinancialDetailsResponse = await(service.getDataFromFinancialServiceForVATVCN("123456789", LocalDate.now(), LocalDate.now()))
+      result.isRight shouldBe true
+      result.right.get shouldBe GetFinancialDetailsSuccessResponse(mockGetFinancialDetailsResponseAsModel)
+    }
+  }
```

Comment:
Is it worth adding some more tests here? Failure scenarios? I've seen in the IT's that this is the case but it may be worth adding unit tests for them as well?